version: "3.7"

services:
  nginx:
    image: nginx:1.19.5
    ports:
      - 80:80
    networks:
      - gis_test_of_test
    volumes:
      - /home/tjalswhd113/nginx.conf:/etc/nginx/nginx.conf
      - gis_test_of_test_static:/data/static/
      - gis_test_of_test_media:/data/media/

  django:
    image: gis_test_of_test:4
    networks:
      - gis_test1
    volumes:
      - gis_test1_static:/home/gis_test_of_test/staticfiles/
      - gis_test1_media:/home/gis_test_of_test/media/
    secrets:
      - DJANGO_SECRET_KEY

  mariadb:
    image: mariadb:10.5
    networks:
      - gis_test_of_test
    volumes:
      - gis_test_of_test_mariadb:/var/lib/mysql
    environment:
      MARIADB_ROOT_PASSWORD: password1234
      MARIADB_DATABASE: django
      MARIADB_USER: django
      MARIADB_PASSWORD: password1234

networks:
  gis_test_of_test:

volumes:
  gis_test_of_test_static:
  gis_test_of_test1_media:
  gis_test_of_test_mariadb:

secrets:
  DJANGO_SECRET_KEY:
    external: true